cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/TestCommand.java

```diff
@@ -336,19 +351,39 @@ public void execute() {
         // Check package files are modified after last build
         boolean isPackageModified = isProjectUpdated(project);
 
-        TaskExecutor taskExecutor = new TaskExecutor.TaskBuilder()
-                .addTask(new CleanTargetCacheDirTask(), isSingleFile) // clean the target cache dir(projects only)
-                .addTask(new ResolveMavenDependenciesTask(outStream)) // resolve maven dependencies in Ballerina.toml
+        CleanTargetCacheDirTask cleanTargetCacheDirTask = new CleanTargetCacheDirTask();
+        ResolveMavenDependenciesTask resolveMavenDependenciesTask = new ResolveMavenDependenciesTask(outStream);
+        CompileTask compileTask = new CompileTask(outStream, errStream, false, isPackageModified,
+                buildOptions.enableCache());
+        RunTestsTask runTestsTask = new RunTestsTask(outStream, errStream, rerunTests, groupList, disableGroupList,
+                testList, includes, coverageFormat, moduleMap, listGroups, excludes, cliArgs);
+
+        CreateTestExecutableTask createTestExecutableTask = null;
+
+        if (!project.buildOptions().cloud().isEmpty()) {
+            //if cloud flag is set, create the test executable
```

Comment:
```suggestion
            // if cloud flag is set, create the test executable
```

Reply:
Fixed in [40229a4](https://github.com/ballerina-platform/ballerina-lang/pull/42183/commits/40229a40c6e8385fc2d56f004746b7a5494411f1) commit.